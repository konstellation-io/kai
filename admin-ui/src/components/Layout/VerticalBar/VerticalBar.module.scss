.container {
  background-color: black;
  flex-shrink: 0;
  display: flex;
  flex-direction: column;
  padding-right: 15px;
}
