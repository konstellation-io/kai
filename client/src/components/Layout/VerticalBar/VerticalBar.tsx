import React, { FunctionComponent } from 'react';
import styles from './VerticalBar.module.scss';

type Props = {
  children: any;
};

const VerticalBar: FunctionComponent<Props> = ({ children }) => (
  <div className={styles.container}>{children}</div>
);

export default VerticalBar;
