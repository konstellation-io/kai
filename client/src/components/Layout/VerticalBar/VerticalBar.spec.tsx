import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VerticalBar from './VerticalBar';
import '@testing-library/jest-dom/extend-expect';


afterEach(cleanup);

it('Renders VerticalBar without crashing', () => {
  const { container } = render(<VerticalBar>
    <div>A</div>
    <div>B</div>
    <div>C</div>
  </VerticalBar>);
  expect(container).toMatchSnapshot();
});

it('Shows children components', () => {
  const { getByText } = render(<VerticalBar>
    <div>A</div>
    <div>B</div>
    <div>C</div>
  </VerticalBar>);

  expect(getByText('A')).toBeInTheDocument();
  expect(getByText('B')).toBeInTheDocument();
  expect(getByText('C')).toBeInTheDocument();
});
