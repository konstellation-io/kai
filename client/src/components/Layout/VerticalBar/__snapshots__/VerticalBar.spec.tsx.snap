// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders VerticalBar without crashing 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      A
    </div>
    <div>
      B
    </div>
    <div>
      C
    </div>
  </div>
</div>
`;
